lib/ansible/cli/__init__.py

```diff
@@ -192,6 +192,12 @@ def split_vault_id(vault_id):
         ret = tuple(parts)
         return ret
 
+    # TODO: verify that nothing else is passing an argument to
+    # create_new_password, then remove it from this function. If anything is
+    # using it inside the ansible codebase, remove it as an argument. If
+    # external code depends on create_new_password being part of the signature,
+    # then add documentation stating that create_new_password cannot be removed
+    # due to external user code relying on it.
     @staticmethod
     def build_vault_ids(vault_ids, vault_password_files=None,
                         ask_vault_pass=None, create_new_password=None,
```

Comment:
It looks like the only other place in our codebase calling this with `create_new_password` is ./test/units/cli/test_cli.py.

The [Python API](https://docs.ansible.com/ansible/latest/dev_guide/developing_api.html#python-api) is not public and from a quick look I don't see any public code on Github using this method. We could deprecate it for a couple releases using `display.deprecated(msg, version='2.19')`, but seems pretty safe to delete.

Reply:
I have deprecated it and updated the changelog fragment to match.